rhcephcompose/gather/koji.py

```diff
@@ -0,0 +1,72 @@
+from hashlib import md5
+import posixpath
+from rhcephcompose import common_koji
+from rhcephcompose.build import Build
+from rhcephcompose.artifacts import BinaryArtifact, SourceArtifact
+from rhcephcompose.log import log
+
+
+# Koji archive types that represent a Debian source package:
+SOURCE_TYPES = ('tar', 'dsc')
+
+
+def query(profile, tag, whitelist):
+    """
+    Find all the artifacts we will need from a given Koji tag.
+
+    :param profile: Koji profile name, eg "koji"
+    :param tag: Koji tag, "ceph-3.2-xenial"
+    :param whitelist: whitelist of deb package names to save
+    :returns: list of rhcephcompose Build objects
+    """
+    if not common_koji.HAS_KOJI:
+        raise RuntimeError('Please install the koji Python library')
+    # Query koji for all builds in the tag.
+    session = common_koji.get_session(profile)
+    buildinfolist = session.listTagged(tag, inherit=True, latest=True,
```

Comment:
It might get odd later if the assigned value is no longer a list. Return values from calls can change the contract over time, and the loop below might be able to still work. For example, `.listTagged` could decide to return a `set()` and then `buildinfolist` wouldn't be a list anymore. It wouldn't necessarily break anything, but then the refactoring would require to update variables that are using data structure names in them that no longer represent what the data structure is.

Reply:
You're right. I've renamed this variable to just be `buildinfos`.